### PR TITLE
Change UpgradeData -> RecurringData

### PR DESCRIPTION
**Pull request by matt-thomason**
# Description
Changes `UpgradeData` -> `RecurringData`. This lets us get the version number of the upgrade just applied, so we can pass this back to Bolt to track merchant's current Bolt module version.

Fixes: https://app.asana.com/0/941920570700290/1157054368227945/f

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: Setup/RecurringData.php [hunk 1]
```diff
@@ -56,16 +56,17 @@ public function __construct(
     }
 
     /**
-     * Called by magenot on module upgrades. We simply get new values for feature
+     * Called by magento on module upgrades. We simply get new values for feature
```
Comment (roopakv) [inline Setup/RecurringData.php]: magenot was better :P

**Comment by matt-thomason** [comment]
No tests for `UpgradeData.php` to adjust.

File: Setup/RecurringData.php [hunk 2]
```diff
@@ -56,16 +56,17 @@ public function __construct(
...
      *
      * @param ModuleDataSetupInterface $setup
      * @param ModuleContextInterface $context
      */
-    public function upgrade(
+    public function install(
         ModuleDataSetupInterface $setup,
         ModuleContextInterface $context
     ) {
+        $this->_logHelper->addInfoLog("MWT test - " .   $context->getVersion());
```
Comment (roopakv) [inline Setup/RecurringData.php]: nuke please

File: Test/Unit/Model/PaymentTest.php [hunk 3]
```diff
@@ -220,6 +220,27 @@ public function voidPayment_success()
         $this->currentMock->void($this->paymentMock);
     }
 
+    /**
+     * @test
+     */
+    public function voidPayment_skipHookNotification()
```
Comment (roopakv) [inline Test/Unit/Model/PaymentTest.php]: did you mean to add this test as part of this PR?

Comment (matt-thomason) [reply Test/Unit/Model/PaymentTest.php]: Nope I messed up my amend. Gone now.

**Pull request merged**